README.md

```diff
@@ -0,0 +1,46 @@
+## redfour
+
+A small library that implements a binary semaphore using Redis.
+
+## Install
+
+`npm install redfour`
+
+## Usage example
+
+```
+var Lock = require('redfour');
+
+var testLock = new Lock({
+	redis: 'redis://localhost:6381',
+	namespace: 'mylock'
+});
+var id = Math.random();
+var firstlock;
+
+// First, acquire the lock.
+testLock.acquireLock(id, 'arbitrarystate', 60 * 1000 /* 60 sec timeout */ , function(err, res) {
+  if (err) console.log('error acquiring', err);
+	else firstlock = res;
+});
+
+// Another server might be waiting for the lock like this.
+testLock.waitAcquireLock(id, '', 10 * 1000, 10 * 1000, function(err, lock) {
+  if (err) console.log('error wait acquiring', err);
+	else console.log('lock acquired! ', lock);
+});
+
+// When the original lock is released, `waitAcquireLock` is fired on the other server.
+setTimeout(() => {
+	testLock.releaseLock(id, firstlock.index, (err) => {
```

Comment:
What do you think about making the second argument the lock itself? (`firstlock` in this case.) To not concern the client with how the locks are identified internally.

I'd love to make the lock entirely opaque, tbh. Of its fields:
* the client doesn't need `id` since it's the same as they passed in
* the client doesn't need `state`, ditto the above
* the client doesn’t need to know `index`, it could be opaque
* we could indicate `success` simply by passing `null` / non-`null` to the acquisition callback (`!!lock` == success, `!lock` == failure)

But we still have `ttl` then. :\